### PR TITLE
Perform static token validation before logging in

### PR DESCRIPTION
**Pull request by thesadru**
## Summary

It might be neat to validate the token before logging in so the users can get a bit more of a helpful error.

## Checklist

- [x] If code changes were made then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `black .`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by shiftinv** [comment]
Are tokens guaranteed to always be 59 chars? Eventually the timestamps/snowflakes should reach a point where their value results in a 60 char token

**Comment by thesadru** [comment]
I'm guessing we could just check whether the token is >= 59 characters. I wanted it to be a proper validation but I suppose we can make it less strict.

File: disnake/utils.py [hunk 1]
```diff
@@ -340,6 +340,15 @@ def parse_token(token: str) -> Tuple[int, datetime.datetime, bytes]:
     """
     parts = token.split(".")
 
+    # validate the token while accounting for any eventual overflows in the future
+    if 53 >= len(token) >= 50:
```
Comment (Victorsitou) [inline disnake/utils.py:344]:
```suggestion
    if 60 >= len(token) >= 50:
```

**Comment by thesadru** [comment]
After going through the various issues which could arise we have decided on closing this pull request.
Checking for client secrets is still being considered.

**Pull request closed without merging**